service/matching/config.go

```diff
@@ -82,6 +82,7 @@ type (
 		MembershipUnloadDelay                    dynamicconfig.DurationPropertyFn
 		TaskQueueInfoByBuildIdTTL                dynamicconfig.DurationPropertyFnWithTaskQueueFilter
 		PriorityLevels                           dynamicconfig.IntPropertyFnWithTaskQueueFilter
+		AutoEnableSub                            dynamicconfig.TypedSubscribableWithTaskQueueFilter[bool]
```

Comment:
as we discussed: I don't think this needs to be a subscription, just check it whenever you need to.